### PR TITLE
Feature bl 10711 cookie compliance

### PR DESCRIPTION
**Pull request by jerotire**

### CODE REVIEW

File: motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/cookie/UserCookiePreferencesFilter.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            this.userCookiePreferences.setShouldUseAnalytics(false);
+        } else {
+            this.decodeUserCookiePreferenceCookie(userCookiePreferenceCookie);
+            this.userCookiePreferences.setIsSet(true);
+        }
+
+        requestContext.setProperty(USER_COOKIE_PREFERENCE_REQUEST_ATTRIBUTE, this.userCookiePreferences);
+    }
+
+    private Cookie getSessionCookie(ContainerRequestContext requestContext) {
```
Comment (aaronlong) [inline motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/cookie/UserCookiePreferencesFilter.java]: Should this be named something like 'getUserPreferencesCookie'

File: motr-webapp/webapp/src/main/resources/template/cookie-preferences.hbs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                    </div>
+                </fieldset>
+            </div>
+
+            <div>
+                <h2 class="heading-large">Strictly necessary cookies</h2>
+                <p>The MOT reminder service uses cookies for session management and security.</p>
+                <p>They always need to be on.</p>
+                <p>
+                    <a href="/cookies">Find out more about cookies on the MOT reminder service</a>
```
Comment (aaronlong) [inline motr-webapp/webapp/src/main/resources/template/cookie-preferences.hbs]: Please use URL helper here.  For example `{{{url this "/"}}}`

File: motr-webapp/webapp/src/test/resources/template/test-helpersregistered.hbs [hunk 3]
```diff
@@ -0,0 +1 @@
+{{#usdfgdgdgences}}userHasSetCookiePreferences{{/usdfgdgdgences}}
```
Comment (aaronlong) [inline motr-webapp/webapp/src/test/resources/template/test-helpersregistered.hbs]: Should this be removed?

File: motr-webapp/webapp/src/main/resources/template/cookie-preferences.hbs (hunk 2 above)
Comment (jerotire) [reply motr-webapp/webapp/src/main/resources/template/cookie-preferences.hbs]: Missed that one :)

File: motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/cookie/UserCookiePreferences.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    private boolean isSet;
+
+    private boolean analytics;
+
+    public boolean isSet() {
+        return this.isSet;
+    }
+
+    public UserCookiePreferences setIsSet(boolean isSet) {
```
Comment (sjduffey) [inline motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/cookie/UserCookiePreferences.java]: The name `setIsSet` seems odd to me... would something like `isPreferencesCookieSet` / `setIsPreferencesCookieSet` be better?

Minor though, so not a blocker to progressing the ticket if there are no other outstanding changes

File: motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/render/helper/ShouldUseAnalyticsHelper.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    private UserCookiePreferences userCookiePreferences;
+
+    public ShouldUseAnalyticsHelper(UserCookiePreferences userCookiePreferences) {
+        this.userCookiePreferences = userCookiePreferences;
+    }
+
+    @Override
+    public Object apply(Object context, Options options) throws IOException {
+        if (this.userCookiePreferences != null
+                && this.userCookiePreferences.isSet()
```
Comment (sjduffey) [inline motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/render/helper/ShouldUseAnalyticsHelper.java]: Does `isSet` need to be a public method? Could it just be a part of the `shouldUseAnalytics` condition, internal to UserCookiePreferences? 

Or is it used somewhere I haven't seen yet :-p

Comment (sjduffey) [reply motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/render/helper/ShouldUseAnalyticsHelper.java]: So, yeah, it is used in the next file...

Comment (jerotire) [reply motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/render/helper/ShouldUseAnalyticsHelper.java]: I agree, after our chat. It looks like isPresent() is a more suitably named method/property for checking if the preference cookie is set/present in the request. Work for abstracting the cookies in the application will be tackled in future tickets.

File: motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/cookie/UserCookiePreferences.java (hunk 4 above)
Comment (jerotire) [reply motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/cookie/UserCookiePreferences.java]: agree, after our chat. It looks like isPresent() is a more suitably named method/property for checking if the preference cookie is set/present in the request. Work for abstracting the cookies in the application will be tackled in future tickets.

**Pull request merged**